### DISCUSSION TITLE
Compare file type within set of subfolders

### DISCUSSION THREAD

**Discussion by burningT** [WinMerge/winmerge#2006] [Q&A] [answered]
Hi, 
I am looking for a way to compare all "*.hex" files within "Exe" folders across two versions of the same folder structure. The present folder structure is as follows:
- Build\Bootloader\SW_X\Exe\
- Build\Bootloader\SW_X\List\
- Build\Firmware\SW_X\Exe\
- Build\Firmware\SW_X\List\

Looking into all these folders works fine with the following filter file:
> def: exclude
> f: \\.cpp$ ## Match *.cpp files
> d: \\\\*$ ## Subfolders

However, changing the last line to look for "Exe" folders (e.g. "d: \\\\Exe")  does return an empty comparison. Irritatingly, "d: \\\\Firmware" works as expected and returns the comparison for all folders within the Firmware folder.

Is Winmerge 2.16.32 only capable of filtering the first subfolder level or am a missing something?

**Comment by sdottaka** [comment, marked as answer]
You also need to specify the ancestor folder of the Exe folder as shown below. (I was not able to reproduce the behavior where only Firmware is specified and everything in Firmware is compared.)
~~~
def: exclude
f: \.hex$
d: \\Build$
d: \\Bootloader$
d: \\Firmware$
d: \\SW_X$
d: \\Exe$
~~~

**Comment by burningT** [comment]
Thank you for your quick response!

Your recommendation indeed does the trick. My "d: \\\\Firmware" does work weird as I underestimated the "$" and skipped it.

However, this leads to a follow up question as "SW_X" represents a multitude of folders. If I understand correctly, I would need to list all these folders explicitly. This fits my observation, that "d: \\\\*\\\\Exe$" would not do the trick. Unfortunately, that is too hard to maintain in my case.  Is there some solution I am missing?

**Reply by sdottaka** [reply]
The filter pattern is specified using regular expressions, so if "SW_X" actually represents folder names like "SW_1" or "SW_ABC," you can specify it as `d: \\SW_[^\\]*$`.

**Reply by burningT** [reply]
Well, unfortunately the folder name pattern is not that simple and not to distinguish from other folders on its own. The  pattern becomes unique when at least one other folder name is considered by the pattern like "d: \\\\*\\\\Exe$". Unfortunately this exact regex does not seem to work. Does Winmerge support anything like that?

**Reply by sdottaka** [reply]
I think you probably meant `d:\\.*\\Exe$` instead of `d: \\*\\Exe$`. It is possible to set this pattern, but unfortunately you still have to specify the ancestor folder of the Exe folder.

**Reply by burningT** [reply]
OK, so I need to deal with that limitation. I appreciate your support. Thank you!
